File: src/main/java/ms/domwillia/jvmemory/monitor/printer/StorePrinter.java
package ms.domwillia.jvmemory.monitor.printer;

import ms.domwillia.jvmemory.monitor.StackTracker;

public class StorePrinter {

	private static void storePrint(String type, Object val, int var) {
		System.out.printf("%s > store %s \"%s\" in local var %d\n", StackTracker.getHead(), type, val.toString(), var);
	}

	public static void booleanDo(boolean val, int var) {
		storePrint("bool", val, var);
	}

	public static void charDo(char val, int var) {
		storePrint("char", val, var);
	}

	public static void byteDo(byte val, int var) {
		storePrint("byte", val, var);
	}

	public static void shortDo(short val, int var) {
		storePrint("short", val, var);
	}

	public static void intDo(int val, int var) {
		storePrint("int", val, var);
	}

	public static void floatDo(float val, int var) {
		storePrint("float", val, var);
	}

	public static void longDo(long val, int var) {
		storePrint("long", val, var);
	}

	public static void doubleDo(double val, int var) {
		storePrint("double", val, var);
	}

	public static void objectDo(Object val, int var) {
		storePrint("object", val, var);
	}
}

File: src/main/java/ms/domwillia/jvmemory/monitor/printer/StorePrinter.kt
package ms.domwillia.jvmemory.monitor.printer

import ms.domwillia.jvmemory.monitor.StackTracker

class StorePrinter(private val stackTracker: StackTracker) {

    private fun storePrint(type: String, value: Any, index: Int) {
        println("${stackTracker.head} > store $type in '$value' in local var $index")
    }

    fun booleanDo(value: Boolean, index: Int) {
        storePrint("bool", value, index)
    }

    fun charDo(value: Char, index: Int) {
        storePrint("char", value, index)
    }

    fun byteDo(value: Byte, index: Int) {
        storePrint("byte", value, index)
    }

    fun shortDo(value: Short, index: Int) {
        storePrint("short", value, index)
    }

    fun intDo(value: Int, index: Int) {
        storePrint("int", value, index)
    }

    fun floatDo(value: Float, index: Int) {
        storePrint("float", value, index)
    }

    fun longDo(value: Long, index: Int) {
        storePrint("long", value, index)
    }

    fun doubleDo(value: Double, index: Int) {
        storePrint("double", value, index)
    }

    fun objectDo(value: Any, index: Int) {
        storePrint("object", value, index)
    }
}
